plugins/ccgx/fu-ccgx-hid-device.c

```diff
@@ -29,13 +29,30 @@ fu_ccgx_hid_device_detach (FuDevice *device, GError **error)
 				       FU_CCGX_HID_DEVICE_TIMEOUT,
 				       FU_HID_DEVICE_FLAG_NONE,
 				       error)) {
-		g_prefix_error (error, "mfg mode error: ");
+		g_prefix_error (error, "switch to HPI mode error: ");
 		return FALSE;
 	}
+	return TRUE;
+}
+
+static gboolean
+fu_ccgx_hid_device_detach (FuDevice *device, GError **error)
+{
+	if (!fu_ccgx_hid_device_enable_hpi_mode (device, error))
+		return FALSE;
 	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
 	return TRUE;
 }
 
+static gboolean
+fu_ccgx_hid_device_setup (FuDevice *device, GError **error)
+{
+	/* This seems insane... but we need to switch the device from HID
+	 * mode to HPI mode at startup. The device continues to function
+	 * exactly as before and no user-visible effects are noted */
```

Comment:
Is it basically enabling the I2C communication to the device?  If so, I think that this should be un-done after enumerating version and such.  Otherwise it's leaving a comms hole for other processes to abuse the end point more easily with timing no?

Reply:
Apparently there's no drawback to leaving it in HPI mode. We still need to be root to use the HPI device, and I suppose it's even more secure than in HID mode which is open by default on most OS's. In the case of one of the Lenovo hubs the HID mode is kinda useless as the only thing you can use it for is to switch the device into HPI mode....